Docs for Starlette and FastAPI integration. (#5329)
getsentry/sentry-docs: src/platforms/python/guides/fastapi/index.mdx

---
title: FastAPI
redirect_from:
  - /clients/python/integrations/fastapi/
  - /platforms/python/fastapi/
description: "Learn about using Sentry with FastAPI."
---

The FastAPI integration adds support for the [FastAPI Framework](https://fastapi.tiangolo.com/).

## Install

Install `sentry-sdk` from PyPI with the `fastapi` extra:

```bash
pip install --upgrade 'sentry-sdk[fastapi]'
```

## Configure

To configure the SDK, initialize it with the integration before your app has been initialized:

```python
from fastapi import FastAPI

import sentry_sdk
from sentry_sdk.integrations.starlette import StarletteIntegration
from sentry_sdk.integrations.fastapi import FastApiIntegration

sentry_sdk.init(
    dsn="___PUBLIC_DSN___",
    integrations=[
        StarletteIntegration(),
        FastApiIntegration(),
    ],

    # Set traces_sample_rate to 1.0 to capture 100%
    # of transactions for performance monitoring.
    # We recommend adjusting this value in production,
    traces_sample_rate=1.0,
)

app = FastAPI()
```

> Because FastAPI is based on the Starlette framework, you need to enable both integrations `StarletteIntegration` and `FastApiIntegration`.

Reviewer: ```suggestion
Because FastAPI is based on the Starlette framework, you need to enable both `StarletteIntegration` and `FastApiIntegration`.
```

> **Note:** In the future we will make this auto enabling, so you do not need to add the `StarletteIntegration` to your `sentry_sdk.init()` call but the Sentry SDK will detect the presence of Starlette and will setup the integration automatically.

Reviewer: ```suggestion
<Note>

In the future, we will make this auto-enabling, so you won't need to add the `StarletteIntegration` to your `sentry_sdk.init()` call. Instead, the Sentry SDK will detect the presence of Starlette and will set up the integration automatically.

</Note>
```

## Verify

This snippet includes an intentional error, so you can test that everything is working as soon as you set it up:

```python
from fastapi import FastAPI

app = FastAPI()

@app.get("/sentry-debug")
async def trigger_error():
    division_by_zero = 1 / 0

```

Visiting `"/sentry-debug"` will trigger an error that will be captured by Sentry.

## Behavior

- All exceptions leading to an Internal Server Error are reported.

- Request data is attached to all events: **HTTP method, URL, headers, form data, JSON payloads**. Sentry excludes raw bodies and multipart file uploads. Sentry also excludes personally identifiable information (such as user ids, usernames, cookies, authorization headers, IP addresses) unless you set `send_default_pii` to `True`.

> - If a `traces_sample_rate` is set, then also performance information is reported than can be seen in the "Performance" section of the Sentry UI.

Reviewer: ```suggestion
- If a `traces_sample_rate` is set, then performance information is also reported, which you can see on the **Performance** page of [sentry.io](https://sentry.io).
```

## Options

You can pass the following keyword arguments to `FastApiIntegration()`:

- `transaction_style`:

> With this option you can influence how the transactions are named in Sentry. Here is an example:

Reviewer: ```suggestion
With this option, you can influence how the transactions are named in Sentry. For example:
```
